h5p-bildetema/src/components/Words/Words.tsx

```diff
@@ -5,115 +5,167 @@ import { useContentId } from "use-h5p";
+
+        if (currentTopicHasTopicImage) {
+          setIsTopicImageView(true);
+
+          const content = existingContent[0];
+          const params = {
+            ...JSON.parse(content.json_content),
+            currentLanguage,
+          };
+
+          const topicView = H5P.newRunnable(
+            {
+              library: getLibraryName({
+                machineName: content.machine_name,
+                majorVersion: content.major_version,
+                minorVersion: content.minor_version,
+              } as Library),
+              params,
+            },
+            content.content_id,
+            H5P.jQuery(topicRootElement),
+          );
+          setTopicViewInstance(topicView);
+        } else {
+          setIsTopicImageView(false);
+          toggleShowTopicImageView(false);
+        }
 
-      return H5P.newRunnable(
-        {
-          library: getLibraryName(gridViewLibrary),
-          params: {
-            words,
-            showWrittenWords,
+        const gridView = H5P.newRunnable(
+          {
+            library: getLibraryName(gridViewLibrary),
+            params: {
+              words,
+              showWrittenWords,
+            },
           },
-        },
-        contentId,
-        H5P.jQuery(rootElement),
-      );
-    };
+          contentId,
+          H5P.jQuery(gridRootElement),
+        );
+        setGridViewInstance(gridView);
+      };
 
-    setTopicViewInstance(getViewInstance(ref.current));
+      setViewInstances(topicViewRef.current, gridViewRef.current);
+    })();
+    return () => {
+      toggleShowTopicImageView(true);
+    };
 
     // Avoid updating when params changes, because we want to trigger changes in the useEffect below
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [contentId, ref.current]);
+  }, [contentId, topicViewRef, gridViewRef]);
 
   useEffect(() => {
     topicViewInstance?.trigger("change-params", {
       words,
       showWrittenWords,
       currentLanguage,
     });
-
+    gridViewInstance?.trigger("change-params", {
+      words,
+      showWrittenWords,
+      currentLanguage,
+    });
     // Avoid updating when `gridViewInstance` changes, because we don't want to trigger updates to the grid view when it initializes
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [words, showWrittenWords, topicViewInstance, currentLanguage]);
-
-  return <div ref={ref} />;
+  }, [
+    words,
+    showWrittenWords,
+    topicViewInstance,
+    gridViewInstance,
+    currentLanguage,
+  ]);
+
+  return (
+    <>
+      <div className={styles.toggle}>
+        {isTopicImageView && (
+          <Toggle
+            label="Topic view"
+            checked={showTopicImageView}
+            handleChange={toggleShowTopicImageView}
+            id={`topic-view-toggle-${contentId}`}
+          />
+        )}
+      </div>
+      <div
+        ref={topicViewRef}
+        style={!showTopicImageView ? { display: "none" } : {}}
```

Comment:
Could perhaps add a class instead of using inline style, something like

```suggestion
        className={!showTopicImageView && styles.display_none}
```